src/common/utils/IndexKeyUtils.cpp

```diff
@@ -81,6 +81,28 @@ std::string IndexKeyUtils::edgeIndexKey(size_t vIdLen,
   return key;
 }
 
+// static
+std::string IndexKeyUtils::indexFirstKey(PartitionID partId) {
+  int32_t item = (partId << kPartitionOffset) | static_cast<uint32_t>(NebulaKeyType::kIndex);
+
+  std::string key;
+  IndexID indexId = 0;
+  key.append(reinterpret_cast<const char*>(&item), sizeof(int32_t))
+      .append(reinterpret_cast<const char*>(&indexId), sizeof(IndexID));
+  return key;
+}
+
+// static
+std::string IndexKeyUtils::indexLastKey(PartitionID partId) {
+  int32_t item = (partId << kPartitionOffset) | static_cast<uint32_t>(NebulaKeyType::kIndex);
+
+  std::string key;
+  IndexID indexId = INT32_MAX;
```

Comment:
Can we guarantee that index_id is always greater than 0? If not, should use 0xffffffff instead of 0x7ffffff

Reply:
The indexId is logically (0, INT32_MAX), although there is no protection in our code.

Also for the above range, 0x7ffffff is the maximum
Whether it is big-endian storage or little-endian storage, there is no bigger number than this.

In order to avoid ambiguity, the minimum and maximum value is directly stored in characters (‘\0’,'\377')

Reply:
👌🏻I was wrong